tiledb/sm/subarray/subarray.h

```diff
@@ -580,27 +600,24 @@ class Subarray {
       const std::vector<T>& tile_coords, std::vector<uint8_t>* aux) const;
 
   /**
-   * Returns the tile overlap of the subarray.
-   *
-   * The outer vector is indexed by fragment ids and the inner vector is
-   * indexed by range indexes.
+   * Returns the internal `TileOverlap` instance. The caller is responsible
+   * for ensuring that the input indexes are valid.
    *
-   * As an optimization, the underlying data structure may be shared between
-   * `Subarray` instances and their partitioned `Subarray` instances created
-   * by the `SubarrayPartitioner`. The caller must use the
-   * `Subarray::overlap_range_offset` to determine the range index that points
-   * to the starting range of this `Subarray` instance.
+   * @param fragment_idx The fragment index.
+   * @param range_idx The range index.
    */
-  const std::vector<std::vector<TileOverlap>>& tile_overlap() const;
+  inline const TileOverlap* get_tile_overlap(
```

Comment:
Move to `subarray.cc`?

Reply:
- Renamed to `tile_overlap` to remove the "get" wordage here.
- This must stay here to remain `inline` since it is used outside of the class.